openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityRepository.java

```diff
@@ -1421,41 +1424,97 @@ private void validateExtension(T entity) {
-        } else if ("dateTime-cp".equals(customPropertyType)) {
-          formatter = DateTimeFormatter.ofPattern(Objects.requireNonNull(propertyConfig));
-          LocalDateTime dateTime = LocalDateTime.parse(fieldValue.textValue(), formatter);
-          ((ObjectNode) jsonNode).put(fieldName, dateTime.format(formatter));
-        } else if ("time-cp".equals(customPropertyType)) {
-          formatter = DateTimeFormatter.ofPattern(Objects.requireNonNull(propertyConfig));
-          LocalTime time = LocalTime.parse(fieldValue.textValue(), formatter);
-          ((ObjectNode) jsonNode).put(fieldName, time.format(formatter));
-        }
+        validateAndUpdateExtensionBasedOnPropertyType(
+            entity,
+            (ObjectNode) jsonNode,
+            fieldName,
+            fieldValue,
+            customPropertyType,
+            propertyConfig);
       } catch (DateTimeParseException e) {
         throw new IllegalArgumentException(
-            CatalogExceptionMessage.dateTimeValidationError(
-                fieldName, TypeRegistry.getCustomPropertyConfig(entityType, fieldName)));
+            CatalogExceptionMessage.dateTimeValidationError(fieldName, propertyConfig));
       }
-      Set<ValidationMessage> validationMessages = jsonSchema.validate(fieldValue);
+
+      Set<ValidationMessage> validationMessages = jsonSchema.validate(entry.getValue());
       if (!validationMessages.isEmpty()) {
         throw new IllegalArgumentException(
             CatalogExceptionMessage.jsonValidationError(fieldName, validationMessages.toString()));
       }
     }
   }
 
+  private void validateAndUpdateExtensionBasedOnPropertyType(
+      T entity,
+      ObjectNode jsonNode,
+      String fieldName,
+      JsonNode fieldValue,
+      String customPropertyType,
+      String propertyConfig) {
+    DateTimeFormatter formatter;
+    switch (customPropertyType) {
+      case "date-cp":
+        DateTimeFormatter inputFormatter =
+            DateTimeFormatter.ofPattern(propertyConfig, Locale.ENGLISH);
+
+        // Parse the input string into a TemporalAccessor
+        TemporalAccessor date = inputFormatter.parse(fieldValue.textValue());
+
+        // Create a formatter for the desired output format
+        DateTimeFormatter outputFormatter =
+            DateTimeFormatter.ofPattern(propertyConfig, Locale.ENGLISH);
+        jsonNode.put(fieldName, outputFormatter.format(date));
+        break;
+      case "dateTime-cp":
+        formatter = DateTimeFormatter.ofPattern(propertyConfig);
+        LocalDateTime dateTime = LocalDateTime.parse(fieldValue.textValue(), formatter);
+        jsonNode.put(fieldName, dateTime.format(formatter));
+        break;
+      case "time-cp":
+        formatter = DateTimeFormatter.ofPattern(propertyConfig);
+        LocalTime time = LocalTime.parse(fieldValue.textValue(), formatter);
+        jsonNode.put(fieldName, time.format(formatter));
+        break;
+      case "metaEnum":
```

Comment:
lets call it EnumWithDescriptions

Reply:
How are we handling exceptions, what happens if one of these fails parsing?

Reply:
have created methods to handle respective types and their exception -
private void validateAndUpdateExtensionBasedOnPropertyType(
    T entity, ObjectNode jsonNode, String fieldName, JsonNode fieldValue, 
    String customPropertyType, String propertyConfig) {

    switch (customPropertyType) {
        case "date-cp", "dateTime-cp", "time-cp" -> {
            String formattedValue = getFormattedDateTimeField(
                fieldValue.textValue(), customPropertyType, propertyConfig, fieldName);
            jsonNode.put(fieldName, formattedValue);
        }
        case "enumWithDescriptions" -> handleEnumWithDescriptions(
            fieldName, fieldValue, propertyConfig, jsonNode, entity);
        default -> {}
    }
}